python/desc/imsim/atmPSF.py

```diff
@@ -0,0 +1,132 @@
+    def _r0_500(wavelength, L0, seeing):
+        """Returns r0_500 to use to get target seeing."""
+        guess = wavelength*1e-9/(seeing/206265)*0.976
+        result = fsolve(AtmosphericPSF._seeing_resid, guess, (wavelength, L0, seeing))
+        return result[0]
+
+    def _getAtmKwargs(self):
+        ud = galsim.UniformDeviate(self.rng)
+        gd = galsim.GaussianDeviate(self.rng)
+
+        # Use values measured from Ellerbroek 2008.
+        altitudes = [0.0, 2.58, 5.16, 7.73, 12.89, 15.46]
+        # Elevate the ground layer though.  Otherwise, PSFs come out too correlated
+        # across the field of view.
+        altitudes[0] = 0.2
+
+        # Use weights from Ellerbroek too, but add some random perturbations.
+        weights = [0.652, 0.172, 0.055, 0.025, 0.074, 0.022]
+        weights = [np.abs(w*(1.0 + 0.1*gd())) for w in weights]
+        weights = np.clip(weights, 0.01, 0.8)  # keep weights from straying too far.
+        weights /= np.sum(weights)  # renormalize
+
+        # Draw a single common outer scale for all layers from a log normal
+        L0 = 0
+        while L0 < 10.0 or L0 > 100:
+            L0 = np.exp(gd() * 0.6 + np.log(25.0))
+        L0 = [L0 for _ in range(6)]
+
+        # Uniformly draw layer speeds between 0 and max_speed.
+        maxSpeed = 20.0
+        speeds = [ud()*maxSpeed for _ in range(6)]
+        # Isotropically draw directions.
+        directions = [ud()*360.0*galsim.degrees for _ in range(6)]
+
+        # Given the desired seeing500 and randomly selected L0, determine appropriate
+        # r0_500
+        r0_500 = AtmosphericPSF._r0_500(500.0, L0, self.seeing500)
+
+        if self.logger:
+            self.logger.debug("airmass = {}".format(self.airmass))
+            self.logger.debug("seeing500 = {}".format(self.seeing500))
+            self.logger.debug("wlen_eff = {}".format(self.wlen_eff))
+            self.logger.debug("r0_500 = {}".format(r0_500))
+            self.logger.debug("L0 = {}".format(L0))
+            self.logger.debug("speeds = {}".format(speeds))
+            self.logger.debug("directions = {}".format(directions))
+            self.logger.debug("altitudes = {}".format(altitudes))
+            self.logger.debug("weights = {}".format(weights))
+
+        return dict(r0_500=r0_500, L0=L0, speed=speeds, direction=directions,
+                    altitude=altitudes, r0_weights=weights, rng=self.rng,
+                    screen_size=819.2, screen_scale=0.1)
+
+    def _getPSF(self, xPupil=None, yPupil=None):
+        """
+        Return a PSF to be convolved with sources.
+
+        @param [in] xPupil the x coordinate on the pupil in arc seconds
+
+        @param [in] yPupil the y coordinate on the pupil in arc seconds
+        """
+        theta = (xPupil*galsim.arcsec, yPupil*galsim.arcsec)
+        psf = self.atm.makePSF(
+            self.wlen_eff,
+            aper=self.aper,
+            theta=theta,
+            t0=self.t0,
+            exptime=self.exptime)
```

Comment:
@jmeyers314 @rmjarvis When I pass a `GSParams` object here that has a `folding_threshold` less than the default, the `psf` that's returned still has a `folding_threshold` of `0.005`.   I think the `galsim.PhaseScreenPSF` object is not being created with a non-default value.   Could you have a look please?

Reply:
Creating a new `aper` object here causes the pupil plane to be recomputed, and in this case, increased the memory usage so it crashed on my laptop with a `MemoryError` (another ~6GB would do that).  I'm thinking that we need some other approach for handling the stamp size for the atmospheric PSF.  I had tried using a Kolmogorov PSF as a proxy, instead of the double Gaussian PSF I tried earlier.  It worked better than the double Gaussian, but the stamp boundaries were still visible.   For that Kolmogorov PSF, I used the parameters read in from the instance catalog.  Maybe there are better parameters to use that could be derived from the atmospheric PSF once it's made?

Reply:
I realized that my prior attempt to use the Kolmogorov PSF for the stamp calculation neglected setting the `folding_threshold` for the PSF itself.  When I do that, the stamps seem large enough, and the bright star images look ok.  I'll do more testing, but at this point I don't think we need the changes to the `_getPsf` function that I've requested here.

Reply:
I think you could do a Kolmogorov convolved with an Airy and get a pretty good approximation of the required stamp size.  Use the desired gsparams on both components.